### PR TITLE
fix: harden demo rehearsal and e2e pipeline scripts (P2 tech debt)

### LINKED ISSUE #215: P2: e2e_pipeline should start from fresh pipeline DB
Source: Codex review on PR #203, comment 3740716175 (scripts/e2e_pipeline.sh:26).

Claim: The scratch DB uses stable path data/output/pipeline-e2e.sqlite and is never removed/reinitialized. On second run (or different fixture same path), pipeline retains prior pages/documents, skips matching records, exporter sends accumulated rows to OLTP, so run is not isolated.

Reproduction:
```
grep -n 'PIPELINE_DB=' scripts/e2e_pipeline.sh
# PIPELINE_DB="${2:-${PIPELINE_DB:-data/output/pipeline-e2e.sqlite}}"
grep -n 'rm.*PIPELINE_DB\|mktemp.*db\|initialize_database' scripts/e2e_pipeline.sh
# no removal or temp DB logic
```

Verified: mkdir -p $(dirname "$PIPELINE_DB") is present but no rm or init-db; janasunani.pipeline.db.initialize_database never called.

Fix is non-blocking per DELIVERY.md (Tier 3 only, not 14 Aug committed gate). File for strict-mode polish. Should create unique temp DB per run or explicitly require --force/confirm and rm existing file, and call init-db.

Suggested:
```
if [ -e "$PIPELINE_DB" ] && [ "${E2E_ALLOW_REUSE_DB:-0}" != "1" ]; then
  rm -f "$PIPELINE_DB"
  # or fail with hint: set E2E_ALLOW_REUSE_DB=1 or remove manually
fi
uv run --extra pipeline-core janasunani-pipeline init-db --db "$PIPELINE_DB"
```

Refs #203

### LINKED ISSUE #216: P2: demo_rehearsal should check specific closure artifacts
Source: Codex review on PR #203, comment 3740731002 (scripts/demo_rehearsal.sh:407).

Claim: When outputs/findings/ contains any file (e.g. PII or discard finding), the closure gate reports success even if both supported closure files are absent: closure_finding_summary.csv / closure_recording_no_action.csv. This bypasses strict-mode missing-artifact warning.

Reproduction:
```
grep -n -A5 'if \[ -d "outputs/findings"' scripts/demo_rehearsal.sh
# FINDINGS_COUNT="$(find outputs/findings -type f | wc -l ...)"
# if [ "$FINDINGS_COUNT" -gt 0 ]; then ok "findings: ..."
```
janasunani/serving/intelligence.py defines _CLOSURE_ARTIFACT_NAMES = ("closure_recording_no_action.csv", "closure_finding_summary.csv")
and fails closed if not exactly one present. Rehearsal should mirror that: check those filenames directly.

Suggested fix:
```
closure_ok=""
for name in closure_finding_summary.csv closure_recording_no_action.csv; do
  if [ -f "outputs/findings/$name" ]; then closure_ok=1; ok "closure: outputs/findings/$name"; break; fi
done
if [ -z "$closure_ok" ]; then check_artifact "outputs/findings/closure_recording_no_action.csv" "closure findings" 0; fi
```

File per DELIVERY.md non-blocking (artifact presence is warn-soft by default; strict is 13 Aug gate). Refs #203

### LINKED ISSUE #218: P2: demo_rehearsal should honor JANASUNANI_SUPERVISOR_AGGREGATES_DIR
Source: Codex review on PR #203, comment 3740731005 (scripts/demo_rehearsal.sh:417).

Claim: When serving uses split config — JANASUNANI_SUPERVISOR_FINDINGS_DIR for closure and JANASUNANI_SUPERVISOR_AGGREGATES_DIR for workload/spike — rehearsal assigns findings dir as aggregate dir and ignores the actual aggregate setting. Closure CSV can make aggregate gate pass even when workload/spike missing, while valid separate aggregate dir reported absent.

Reproduction:
```
grep -n 'AGG_DIR=' scripts/demo_rehearsal.sh
# AGG_DIR="${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}"
grep -n 'SUPERVISOR.*DIR' janasunani/serving/intelligence.py
# configured = os.environ.get("JANASUNANI_SUPERVISOR_FINDINGS_DIR")
# aggregates_configured = os.environ.get("JANASUNANI_SUPERVISOR_AGGREGATES_DIR")
```
ArtifactSupervisorProvider takes findings_dir + aggregates_dir separately; provider_from_env reads both.

Suggested fix:
```
AGG_DIR="${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}}"
# or explicit check of AGGREGATES_DIR first, then FINDINGS_DIR fallback matching intelligence.py default
```

Note: P1 #3740716173 fix now gates data/ probe behind REHEARSAL_ALLOW_DATA=1; this P2 remains distinct (wrong env var). Filed per DELIVERY.md non-blocking polish.

Refs #203

### LINKED ISSUE #219: P2: Exercise production spam scoring path instead of manufacturing spam.json sidecar
Tech-debt from Codex review on #205 (P2 3740728675).

**Location:** `tests/test_pipeline_e2e.py:400` `_try_run_spam_sidecar` / `tests/test_pipeline_e2e.py:450` `test_spam_sidecar_after_pii_before_page_type_writes_bounded_score`

**Problem:** Test manufactures `spam.json` inside the test helper (`score_spam(combined)` -> write JSON via `sidecar_path.write_text`) and never invokes a production after-PII writer/CLI. When the batch sidecar or its pipeline wiring is broken or removed, the test still passes because the sidecar is fabricated in-test. Reviewer suggested invoking `janasunani-spam-score` / production writer.

**Expected (investigated):** No production batch writer currently exists for this exact artifact-DB hop. `janasunani.pipeline.pipeline.STAGE_ORDER` is 6 stages (no spam stage); spam is a **sidecar + serving triage** per architecture — batch spam scores the lake slice via `janasunani.pipeline.spam:main` (`janasunani-spam-score --slice Sambalpur/2024` writes prevalence JSON over lake, not per-page `spam.json`), while live submissions score via `janasunani/serving/triage.py:UnwiredTriageProvider.assess` -> `score_spam` (advisory). The helper already exercises production scorer (`from janasunani.pipeline.spam import score_spam`) — the fabricated `spam.json` is the test's own file assertion, not a reimplementation of a missing production file writer.

**Impact:** Low — synthetic seventh-hop contract, not blocking 14 Aug (P2). The real demo gates are lake prevalence + live triage numeric score, both already wired elsewhere (`tests/test_demo_integration.py` verifies live `triage.spam.spam_score` via processor). Filing to track whether a production batch sidecar writer should exist.

**Repro:**
```bash
grep -n "STAGE_ORDER\|spam" janasunani/pipeline/pipeline.py  # -> 6 stages, no spam
grep -n "def main" janasunani/pipeline/spam.py  # -> CLI scores lake slice, not artifact DB per-page sidecar
grep -n "score_spam" janasunani/serving/triage.py  # -> live path already uses score_spam
```

**Fix options (if prioritized):**
- **Option A (no batch file):** Keep current scorer invocation; rename docstring to clarify that `spam.json` is a test probe, not a production artifact, and assert via `UnwiredTriageProvider` / `score_spam_review` directly instead of file existence.
- **Option B (add production writer):** Introduce a batch writer (e.g. `janasunani.pipeline.spam:write_sidecar` or CLI flag `--write-artifact-sidecar`) that scores `pages.redacted_text` after PII and writes the per-page sidecar the test currently fabricates, then have the test invoke that writer/CLI over the synthetic artifact DB. This is design work, not a one-line test fix.

Filed per CONTRIBUTING.md + agent-conventions (valid observation, remedy is separate claim and not cheap; prior #204 P2s filed as issues for similar design-gap).

Refs #205, tests/test_pipeline_e2e.py:400, janasunani/pipeline/spam.py:440

### PR DESCRIPTION
**Pull request by ymohanty**
## Summary

Four P2 tech-debt findings against the 14 Aug demo rehearsal gate (Codex review on #203/#205), all in `scripts/demo_rehearsal.sh` / `scripts/e2e_pipeline.sh` and their covering tests:

- **#215** — `e2e_pipeline.sh` reused a stale `PIPELINE_DB` across runs (every stage skipped already-processed rows, the exporter re-upserted old rows into OLTP). Added a guard that only auto-deletes the script's own default scratch path; anything else needs `E2E_ALLOW_DB_RESET=1` (or `E2E_ALLOW_REUSE_DB=1` to keep it as-is), then calls `janasunani-pipeline init-db`.
- **#216** — `demo_rehearsal.sh`'s closure-artifact check passed on *any* file under `outputs/findings/`, even ones `janasunani/serving/intelligence.py`'s closure reader would never accept. Now checks specifically for `closure_finding_summary.csv` / `closure_recording_no_action.csv` (`_CLOSURE_ARTIFACT_NAMES`).
- **#218** — `demo_rehearsal.sh` read `JANASUNANI_SUPERVISOR_FINDINGS_DIR` for the aggregates (workload/spike) check, ignoring `JANASUNANI_SUPERVISOR_AGGREGATES_DIR` entirely. `AGG_DIR` now prefers `AGGREGATES_DIR`, falling back to `FINDINGS_DIR`, mirroring `ArtifactSupervisorProvider`'s own fallback.
- **#219** — `tests/test_pipeline_e2e.py`'s spam-sidecar test fabricated `spam.json` in-test (called `score_spam()` directly, wrote the file, asserted on the file it just wrote) — the gate stayed green even if the live triage wiring were broken. Rewrote the helper to route through `janasunani.serving.triage.UnwiredTriageProvider` (the same path live submissions and `demo_rehearsal.sh`'s own `triage.spam.spam_score` check use), failing loudly on `advisory_provider_unavailable` instead of a fabricated fallback.

Combined into one PR: all four findings sit in the same two scripts (plus their covering tests) and are small enough individually that splitting would mean four near-empty PRs touching files the others also touch.

## Safety verification

- `#215`'s destructive path (`rm -f "$PIPELINE_DB"`) can never target a Postgres DSN — the pipeline CLI's `--db` is always a `Path` (sqlite file), never a connection string. The guard was verified by running the real script (subprocess, copied into a scratch dir) against a non-default stale DB: it refuses and exits non-zero, and the file is left byte-for-byte untouched. The default-path/opt-in reset path was verified by calling the real `uv run --extra pipeline-core janasunani-pipeline init-db` against a throwaway sqlite file.
- `#219`'s fix was verified by deliberately breaking the production import in `janasunani/serving/triage.py` (forced an exception before the `score_spam` import): the test went red with the new assertion (`production spam scorer unavailable via UnwiredTriageProvider.assess`). Reverted and confirmed green again.
- `#216`/`#218` were verified by extracting the real `phase_c_artifacts` function body from the script and running it against real scratch directories (an unrelated file only → warns; a real `closure_finding_summary.csv` → ok; `JANASUNANI_SUPERVISOR_AGGREGATES_DIR` set alongside a different `JANASUNANI_SUPERVISOR_FINDINGS_DIR` → the aggregates dir wins).
- No test in this PR touches the production Postgres or `data/`; the `#215` functional tests never resolve OLTP at all, and fail before any `uv run --extra pipeline-core janasunani-pipeline run` invocation.

## Test plan

- [x] `uv run ruff check .` — all checks passed
- [x] `uv run --extra serving --extra pipeline-core pytest` — 1177 passed, 19 skipped (pre-existing skips), 0 failed
- [x] `bash -n scripts/demo_rehearsal.sh` / `bash -n scripts/e2e_pipeline.sh` — both valid
- [x] Manual real-code-path verification of each fix (see Safety verification above)

Fixes #215
Fixes #216
Fixes #218
Fixes #219
---

## Merge note — Codex review round skipped

Merged without a further Codex round at the repository owner's explicit direction: *"the codex reviews keep returning issues each time without converging ... please then merge it idc"*.

Recording this plainly rather than implying a policy basis: `CONTRIBUTING.md` scopes the `codex-review-not-required` label to config-only branches and to Codex being out of review credits. **Neither applies here.** This is a deliberate override by the repository owner, labelled so the exception is visible in the record.

All findings raised on this PR were replied to in-thread and resolved. Any finding not fixed here is filed as a tracked issue rather than dropped.

### CODE REVIEW

**Comment by ymohanty** [comment]
@codex review

File: scripts/demo_rehearsal.sh [hunk 1]
```diff
@@ -403,38 +403,49 @@ phase_c_artifacts() {
...
-    else
-      check_artifact "outputs/findings/closure_finding_summary.csv" "closure findings (outputs/findings/ empty)" 0
+    ls -1 outputs/findings | head -20 | sed 's/^/    /'
+  fi
+  closure_ok=""
+  for name in closure_finding_summary.csv closure_recording_no_action.csv; do
+    if [ -f "outputs/findings/$name" ]; then
+      ok "closure: outputs/findings/$name"
+      closure_ok=1
+      break
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh]: Fixed in `fix(rehearsal): match the provider's own availability rules in phase C`.

Correct, and it is the failure mode a rehearsal gate exists to avoid: the gate said available while `_load_closure()` would have reported unavailable, because it requires exactly one candidate and the loop stopped at the first match. The transition between the two filenames is exactly when a gate matters, and that was the window where it lied.

Candidates are counted now, and two is a failure with the count named so the operator can see which state they are in.

Tests execute the real `phase_c_artifacts` body against scratch directories rather than asserting on the script text, so they check behaviour: one artifact accepted, two rejected.

File: scripts/demo_rehearsal.sh [hunk 2]
```diff
@@ -403,38 +403,49 @@ phase_c_artifacts() {
...
+    check_artifact "outputs/findings/closure_finding_summary.csv" "closure findings (neither closure_finding_summary.csv nor closure_recording_no_action.csv present)" 0
   fi
 
-  # 3. Aggregates — explicit JANASUNANI_SUPERVISOR_FINDINGS_DIR only; data/ probe requires opt-in per AGENTS.md
-  AGG_DIR="${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}"
+  # 3. Aggregates — explicit JANASUNANI_SUPERVISOR_AGGREGATES_DIR, falling
+  #    back to JANASUNANI_SUPERVISOR_FINDINGS_DIR (mirrors
+  #    ArtifactSupervisorProvider's own fallback in intelligence.py); data/
+  #    probe requires opt-in per AGENTS.md
+  AGG_DIR="${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}}"
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh:481]: Fixed in the same commit.

Correct. The gate read only whichever variable was set first, so an aggregates dir that was set but absent or empty failed a strict run even when the findings dir held valid `workload.csv` and `spike.csv` and the panels would have rendered fine. That is the mirror image of the closure bug: the gate failing a demo that would have worked, rather than passing one that would not.

Both directories are now inspected, in the provider's own order `(aggregates_dir, findings_dir)`. Two tests cover it: the findings fallback honoured when aggregates is empty, and aggregates preferred when it has content.

**Comment by ymohanty** [comment]
@codex review

File: scripts/demo_rehearsal.sh [hunk 3]
```diff
@@ -403,38 +403,72 @@ phase_c_artifacts() {
...
-
-  # 3. Aggregates — explicit JANASUNANI_SUPERVISOR_FINDINGS_DIR only; data/ probe requires opt-in per AGENTS.md
-  AGG_DIR="${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}"
-  if [ -n "$AGG_DIR" ] && [ -d "$AGG_DIR" ]; then
-    AGG_COUNT="$(find "$AGG_DIR" -type f -name "*.csv" | wc -l | tr -d ' ')"
-    if [ "$AGG_COUNT" -gt 0 ]; then
-      ok "aggregates: $AGG_DIR ($AGG_COUNT csv(s))"
-    else
-      check_artifact "$AGG_DIR/*.csv" "aggregates in JANASUNANI_SUPERVISOR_FINDINGS_DIR" 0
+    check_artifact "outputs/findings/closure_finding_summary.csv" "closure findings ($closure_count candidates present; the provider requires exactly one and reports unavailable otherwise)" 0
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh]: Fixed in `fix(rehearsal): emit real failures instead of routing through check_artifact`.

Correct, and worth being explicit about what happened: the previous round fixed the *detection* (counting candidates instead of stopping at the first match) but left the *failure path* routed through `check_artifact`, which is existence-based. Because `outputs/findings/closure_finding_summary.csv` genuinely exists in the duplicate case, `check_artifact` reported OK and incremented nothing — so the count was computed correctly and then thrown away. The gate still passed a demo where `ArtifactSupervisorProvider._load_closure()` would report unavailable (`len(candidates) != 1`).

Fixed by calling `fail()` directly from the branch that already knows `closure_count == 2`, with no helper in between:

```bash
else
  fail "closure findings ($closure_count candidates present; the provider requires exactly one and reports unavailable otherwise)"
fi
```

This is now an unconditional failure, not gated by `--strict` — a duplicate is a deterministic break (the provider will refuse to serve it), not a "not published yet" state that only matters in strict mode.

Test: `tests/test_deploy_stack.py::test_phase_c_rejects_two_closure_artifacts` now asserts `FAILURES=1` in the captured output, not just the log message. It runs against the real `check_artifact` (previously the test harness mocked `check_artifact` to always fail, which is exactly what was masking this bug — the assertion passed regardless of what the real helper would have done). Ran locally: `uv run --extra serving --extra pipeline-core pytest tests/test_deploy_stack.py -q -k phase_c` → 6 passed.

File: scripts/demo_rehearsal.sh [hunk 4]
```diff
@@ -403,38 +403,72 @@ phase_c_artifacts() {
...
+  AGG_FOUND=""
+  AGG_COUNT=0
+  for candidate in "${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-}" "${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}"; do
+    [ -n "$candidate" ] || continue
+    [ -d "$candidate" ] || continue
+    candidate_count="$(find "$candidate" -type f -name "*.csv" | wc -l | tr -d ' ')"
+    if [ "$candidate_count" -gt 0 ]; then
+      AGG_FOUND="$candidate"
+      AGG_COUNT="$candidate_count"
+      break
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh]: Fixed in the same commit: `fix(rehearsal): emit real failures instead of routing through check_artifact`.

Correct. The check counted any `*.csv` in the candidate directory, so a directory holding only `workload.csv`, or an unrelated csv, reported "aggregates available" even though `ArtifactSupervisorProvider._load_spike()` looks specifically for `spike.csv` and would report that panel unavailable. And since this was a newly-added recognition of the aggregates directory, that configuration flipped from a strict failure to a silent pass.

`workload.csv` and `spike.csv` are now searched independently, each walking `(JANASUNANI_SUPERVISOR_AGGREGATES_DIR, JANASUNANI_SUPERVISOR_FINDINGS_DIR)` in that order — the same two-directory fallback `_load_workload()` / `_load_spike()` each use for their own named file:

```bash
WORKLOAD_DIR=""
for candidate in "${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-}" "${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}"; do
  [ -n "$candidate" ] || continue
  if [ -f "$candidate/workload.csv" ]; then WORKLOAD_DIR="$candidate"; break; fi
done
# same pattern for SPIKE_DIR / spike.csv
```

Both must be found (independently, possibly in different directories) to pass. A directory that is configured but missing one of the two named files fails outright (not gated by `--strict`), since that is a deterministic break rather than a "nothing published yet" state.

Four behaviours pinned in `tests/test_deploy_stack.py`:
- `test_phase_c_fails_when_aggregates_has_only_workload_csv` — aggregates dir with only `workload.csv`, no `spike.csv` anywhere → fails (`missing: spike.csv`).
- `test_phase_c_fails_on_unrelated_csv_in_aggregates_dir` — an unrelated csv alone → fails (`missing: workload.csv, spike.csv`).
- `test_phase_c_workload_in_aggregates_spike_in_findings_passes` — `workload.csv` in aggregates, `spike.csv` in findings → passes, exercising the per-artifact fallback.
- `test_phase_c_falls_back_to_findings_when_aggregates_is_empty` — updated to assert both names resolve from the findings dir.

Ran locally: `uv run --extra serving --extra pipeline-core pytest tests/test_deploy_stack.py -q -k phase_c` → 6 passed. Full gate also green: `uv run ruff check .` and `JANASUNANI_MODELS_DIR=/nonexistent uv run --extra serving --extra pipeline-core pytest` → 1183 passed, 19 skipped.

**Comment by ymohanty** [comment]
@codex review

File: scripts/demo_rehearsal.sh [hunk 5]
```diff
@@ -403,38 +403,93 @@ phase_c_artifacts() {
...
+    [ -n "$candidate" ] || continue
+    if [ -f "$candidate/spike.csv" ]; then
+      SPIKE_DIR="$candidate"
+      break
+    fi
+  done
+  AGG_DIR="${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}}"
+  if [ -n "$WORKLOAD_DIR" ] && [ -n "$SPIKE_DIR" ]; then
+    if [ "$WORKLOAD_DIR" = "$SPIKE_DIR" ]; then
+      ok "aggregates: $WORKLOAD_DIR (workload.csv, spike.csv)"
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh:484]: Fixed in `fix(rehearsal): require findings dir before passing aggregates`.

Confirmed real. Reproduced before fixing: with `JANASUNANI_SUPERVISOR_AGGREGATES_DIR` pointed at a directory holding both `workload.csv` and `spike.csv`, and `JANASUNANI_SUPERVISOR_FINDINGS_DIR` unset, the rehearsal printed `[ OK ] aggregates: .../aggregates (workload.csv, spike.csv)` with `FAILURES=0`. But `supervisor_provider_from_env()` gates the whole aggregate seam on `JANASUNANI_SUPERVISOR_FINDINGS_DIR` alone -- if it's unset, it returns `UnavailableSupervisorProvider` before `ArtifactSupervisorProvider` is ever constructed, so the aggregates dir's contents are never consulted live. The rehearsal was searching the aggregates dir directly instead of mirroring that on/off switch, so it passed a config the live provider would refuse outright.

Fixed by mirroring the provider's own enablement gate before any per-file lookup: if `JANASUNANI_SUPERVISOR_AGGREGATES_DIR` is set but `JANASUNANI_SUPERVISOR_FINDINGS_DIR` is not, the check now fails immediately (not gated by `--strict` -- same category as the sibling findings on this PR, a deterministic break rather than a "nothing published yet" state):

```bash
if [ -z "${JANASUNANI_SUPERVISOR_FINDINGS_DIR:-}" ] && [ -n "${JANASUNANI_SUPERVISOR_AGGREGATES_DIR:-}" ]; then
  fail "JANASUNANI_SUPERVISOR_AGGREGATES_DIR is set but JANASUNANI_SUPERVISOR_FINDINGS_DIR is not; supervisor_provider_from_env() requires FINDINGS_DIR to enable the aggregate seam at all, so these aggregates would never be served"
else
  ... existing per-file two-directory fallback search unchanged ...
fi
```

Test: `tests/test_deploy_stack.py::test_phase_c_fails_when_findings_dir_unset_even_with_complete_aggregates` runs the real `phase_c_artifacts` body (not a stub) against a scratch aggregates dir with both named files present and `JANASUNANI_SUPERVISOR_FINDINGS_DIR` unset, and asserts `FAILURES=1` and that the old vacuous "aggregates: ... (workload.csv, spike.csv)" pass line no longer appears. The two existing per-file-naming tests (`test_phase_c_fails_when_aggregates_has_only_workload_csv`, `test_phase_c_fails_on_unrelated_csv_in_aggregates_dir`) now set an empty `JANASUNANI_SUPERVISOR_FINDINGS_DIR` so they keep exercising the naming check they were written for, rather than being intercepted by this new gate.

Ran locally: `uv run --extra serving --extra pipeline-core pytest tests/test_deploy_stack.py -q -k phase_c` -> 6 passed. Full suite: 1184 passed, 19 skipped.

**Comment by ymohanty** [comment]
@codex review

File: scripts/demo_rehearsal.sh [hunk 6]
```diff
@@ -403,40 +403,108 @@ phase_c_artifacts() {
...
+  fi
+  # ArtifactSupervisorProvider._load_closure() reports unavailable unless
+  # EXACTLY ONE candidate exists (len(candidates) != 1), so stopping at the
+  # first match let a strict rehearsal pass while the live closure panel was
+  # unavailable -- precisely the transition state between the two filenames.
+  # Count them instead of short-circuiting.
+  closure_found=""
+  closure_count=0
+  for name in closure_finding_summary.csv closure_recording_no_action.csv; do
+    if [ -f "outputs/findings/$name" ]; then
```
[the remark this answers is not in the document]
Comment (ymohanty) [reply scripts/demo_rehearsal.sh:423]: Confirmed real, filed as #245 rather than fixed here. `[ -f ]` follows symlinks and `_is_regular_artifact()` rejects them, so a symlinked artifact makes the rehearsal pass while the live panel is unavailable — the exact vacuous-pass shape this PR set out to remove, one layer down.

Not fixed in this PR because #231 is merging without a further review round at the repository owner's direction. Nothing in the current publishing path creates symlinks, so it is latent rather than live.
